File: worktree-pipeline/src/main/java/com/smartwallet/repository/WalletRepository.java
package com.smartwallet.repository;

import com.smartwallet.entity.Wallet;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface WalletRepository extends JpaRepository<Wallet, Long> {
    List<Wallet> findByUserId(Long userId);
    Optional<Wallet> findByIdAndUserId(Long id, Long userId);
    boolean existsByIdAndUserId(Long id, Long userId);
    long countByUserId(Long userId);
}

File: src/main/kotlin/com/smartwallet/portfolio/repository/WalletRepository.kt
package com.smartwallet.portfolio.repository

import com.smartwallet.portfolio.model.Wallet
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WalletRepository : JpaRepository<Wallet, Long> {
    fun findAllByUserId(userId: Long): List<Wallet>
}
